### PR TITLE
multi server pod failure Test-40582, 40584

### PR DESCRIPTION
**Pull request by rahul-hatwar**
Signed-off-by: Rahul Hatwar <rahulchandrakant.hatwar@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by rahul-hatwar** [comment]
s']}, {'nodeid': 'tests/s3/versioning/test_versioning_put_object.py::TestVersioningPutObject::test_put_object_versioning_suspended_32733', 'test_id': 'TEST-32733', 'marks': ['s3_ops']}, {'nodeid': 'tests/security/test_cortx_port_scanner_kubectl_svc.py::test_cortx_port_scanner_kubectl_svc', 'test_id': 'TEST-34217', 'marks': ['security']}, {'nodeid': 'tests/security/test_cortx_port_scanner_netstat.py::test_cortx_port_scanner_netstat', 'test_id': 'TEST-34218', 'marks': ['security']}] created at /root/rahul/cortx-test-1/log/te_meta.json
Successfully unmounted directory

========================================== 2083 tests collected in 4.98s ==========================================

File: tests/ha/test_multi_server_pod_failure.py [hunk 1]
```diff
@@ -629,3 +630,354 @@ def test_writes_till_kserver_pods_fail(self):
...
+        download_checksum = self.ha_obj.cal_compare_checksum(file_list=[download_path],
+                                                             compare=False)[0]
+        assert_utils.assert_equal(upload_checksum, download_checksum,
+                                  f"Failed to match checksum: {upload_checksum},"
+                                  f" {download_checksum}")
+        LOGGER.info("Matched checksum: %s, %s", upload_checksum, download_checksum)
+        LOGGER.info("Step 10: Successfully downloaded the object and verified the checksum")
+
+        LOGGER.info("Step 11: Perform WRITEs-READs-Verify-DELETEs with variable object sizes. 0B "
+                    "+ (1KB - 512MB) on degraded cluster")
```
Comment (ankita-93) [inline tests/ha/test_multi_server_pod_failure.py:972]: Cluster won't be in degraded state.

File: tests/ha/test_multi_server_pod_failure.py [hunk 2]
```diff
@@ -629,3 +630,358 @@ def test_writes_till_kserver_pods_fail(self):
...
+        assert_utils.assert_true(resp[0], resp[1])
+        download_checksum = self.ha_obj.cal_compare_checksum(file_list=[download_path],
+                                                             compare=False)[0]
+        assert_utils.assert_equal(upload_checksum, download_checksum,
+                                  f"Failed to match checksum: {upload_checksum},"
+                                  f" {download_checksum}")
+        LOGGER.info("Matched checksum: %s, %s", upload_checksum, download_checksum)
+        LOGGER.info("Step 10: Successfully downloaded the object and verified the checksum")
+
+        LOGGER.info("Step 11: Perform WRITEs-READs-Verify-DELETEs with variable object sizes. 0B "
```
Comment (sonalk0209) [inline tests/ha/test_multi_server_pod_failure.py:975]: remove 0B sizes from the file

**Review by sonalk0209 — APPROVED** [review]
@rahul-hatwar minor comment, please check

File: tests/ha/test_multi_server_pod_failure.py [hunk 3]
```diff
@@ -629,3 +630,357 @@ def test_writes_till_kserver_pods_fail(self):
...
+        LOGGER.info("COMPLETED: Test to verify multipart upload after all K server pods "
+                    "down - unsafe shutdown.")
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.skip(reason="Multipart feature F-20C targeted for PI-7")
+    @pytest.mark.tags("TEST-40584")
+    @CTFailOn(error_handler)
+    def test_part_mpu_after_kserver_pods_fail(self):
```
Comment (ApurwaMohite) [inline tests/ha/test_multi_server_pod_failure.py:787]: till_kserver_pods_fail

**Review by ApurwaMohite — APPROVED** [review]

**Review by ankita-93 — APPROVED** [review]

**Pull request merged**